### ISSUE TITLE
High disk usage and cpu spikes when using some sites.

### ISSUE THREAD

**Issue by bitraid** [qutebrowser/qutebrowser#3519]
This started happening after 5605d3cd8.
A site that this is very obvious, and i happen to use all the time, is [inoreader.com](http://inoreader.com) (online rss reader).
When browsing the feeds, or even worse, when browsing folders, there is high disk activity and qutebrowsers'  cpu usage gets very high.
There is an [inoreader demo page ](http://inoreader.com/u/demo), where you can try this out, if you don't have an account there:
"Click" on "Search or subscribe" text box and select a category, select a source, switch to passthrough or insert mode, press `j` to browse the feeds.

**Comment by The-Compiler** [comment]
The commit you linked is a merge from `master` into some branch - how did you get to that commit?

Also, I get roughly the same CPU usage with `python3 testbrowser_webengine.py http://www.inoreader.com/u/demo` - I'm assuming you don't when you say it happened only after that commit?

**Comment by bitraid** [comment]
Isn't it the `webengine_caret` merge? 63658d3a1 and previous worked fine.

**Comment by bitraid** [comment]
Also the problem is much more obvious when browsing through folders with `shift+j`, but this can't be tested on the demo page.

**Comment by The-Compiler** [comment]
Both of those commits are commits which you should never have been on if you're tracking the `master` branch. They're both from the PR branch, before caret browsing was ever merged into master.

Can you please find the commit causing this in the master branch (ideally using git bisect)? Also, can you test with the testbrowser like mentioned above?

**Comment by bitraid** [comment]
Yes, sure. I'll do that later today.

**Comment by bitraid** [comment]
Sorry i messed up before, the problem seems to be with 73b6dea.
The strange thing is, that with testbrowser, i can't reproduce it.
Also, the latest Windows build seems to be working fine.
I'm testing on Linux: 4.14.14-1-ARCH x64, QT: 5.10.0, pyQT: 5.9.2, Python: 3.6.4

**Comment by The-Compiler** [comment]
Looks like that with that workaround (for a Qt bug), we get a "finished loading" event every time we switch between articles, which causes it to be inserted into history and some other stuff to happen:

```
18:14:37 DEBUG    misc       app:on_focus_object_changed:833 Focus object changed: <PyQt5.QtWidgets.QWidget object at 0x7fb40c09fa68>
18:14:38 DEBUG    sql        sql:__init__:150 Preparing SQL query: "INSERT INTO History (url, title, atime, redirect) values(:url, :title, :atime, :redirect)"
18:14:38 DEBUG    sql        sql:run:168 Running SQL query: "INSERT INTO History (url, title, atime, redirect) values(:url, :title, :atime, :redirect)"
18:14:38 DEBUG    sql        sql:run:171 query bindings: {':atime': 1516727678, ':redirect': False, ':title': 'Inoreader - Amazon doesn’t care if you accidentally shoplift from its cashier-less store', ':url': 'http://www.inoreader.com/article/3a9c6e7edbfb100a-amazon-doesnt-care-if-you-accidentally-shoplift-from-its-cashier-less-store'}
18:14:38 DEBUG    sql        sql:__init__:150 Preparing SQL query: "REPLACE INTO CompletionHistory (url, title, last_atime) values(:url, :title, :last_atime)"
18:14:38 DEBUG    sql        sql:run:168 Running SQL query: "REPLACE INTO CompletionHistory (url, title, last_atime) values(:url, :title, :last_atime)"
18:14:38 DEBUG    sql        sql:run:171 query bindings: {':last_atime': 1516727678, ':title': 'Inoreader - Amazon doesn’t care if you accidentally shoplift from its cashier-less store', ':url': 'http://www.inoreader.com/article/3a9c6e7edbfb100a-amazon-doesnt-care-if-you-accidentally-shoplift-from-its-cashier-less-store'}
18:14:38 DEBUG    sessions   sessions:save:289 Saving session _autosave to /tmp/qutebrowser-basedir-l8ardn2q/data/sessions/_autosave.yml...
18:14:38 DEBUG    webview    browsertab:_set_load_status:692 load status for <qutebrowser.browser.webengine.webenginetab.WebEngineTab tab_id=0 url='http://www.inoreader.com/article/3a9c6e7edbfb100a-amazon-doesnt-care-if-you-accidentally-shoplift-f…'>: LoadStatus.success
18:14:38 DEBUG    signals    signalfilter:_filter_signals:86 emitting: cur_load_status_changed('success') (tab 0)
18:14:38 DEBUG    signals    signalfilter:_filter_signals:86 emitting: cur_load_finished(True) (tab 0)
```

Can you try whether the changes in #3509 help?

**Comment by bitraid** [comment]
Yes, that makes things better.

**Comment by The-Compiler** [comment]
Closing this then as I don't think there's anything more we can do there. When [QTBUG-65223](https://bugreports.qt.io/browse/QTBUG-65223) is fixed, it should go away as well as we can disable the workaround again.

**Comment by Aster89** [comment]
I keep observing an increase in CPU usage as well. Specifically I notice a correlation with the use of Microsoft Teams online.

This makes the temperature rise a lot (sometimes it surpasses ~100°C).

However I also observe an "obscillating" trend in the fan speed.

Here's the top of my `top`:
```
Tasks: 208 total,   2 running, 206 sleeping,   0 stopped,   0 zombie
%Cpu(s):  3.4 us,  2.8 sy,  0.0 ni, 85.0 id,  0.0 wa,  8.6 hi,  0.3 si,  0.0 st
MiB Mem :   7871.2 total,   2453.7 free,   1842.4 used,   3575.1 buff/cache
MiB Swap:  16384.0 total,  16384.0 free,      0.0 used.   4735.6 avail Mem 

    PID USER      PR  NI    VIRT    RES    SHR S  %CPU  %MEM     TIME+ COMMAND                                                                                                                                     
 231713 enrico    20   0 3064628 590612 102196 S  13.0   7.3   7:23.55 QtWebEngineProc                                                                                                                             
 241163 root      20   0       0      0      0 R  11.6   0.0   0:48.53 kworker/0:2-kacpi_notify                                                                                                                    
 231380 enrico    20   0 3211808 525960 206052 S  10.0   6.5   2:57.88 qutebrowser                                                                                                                                 
    572 enrico    20   0  760500  98220  63588 S   6.6   1.2  69:15.27 Xorg                                                                                                                                        
    645 enrico     9 -11 1960144  15152  10996 S   4.7   0.2  52:10.11 pulseaudio                                                                                                                                  
 242184 enrico    20   0  516336  52796  39008 S   4.0   0.7   0:18.07 pavucontrol                                                                                                                                 
    671 enrico    20   0  662544  13028   7464 S   1.3   0.2  29:42.67 conky                                                                                                                                       
  31527 enrico    20   0   20540  12968  10344 S   1.0   0.2   7:58.75 i3bar                                                                                                                                       
  38438 enrico    20   0 2394612 253060  91568 S   1.0   3.1  22:57.99 remmina                                                                                                                                     
 241894 enrico    20   0 1275464  56776  47880 S   1.0   0.7   0:03.29 QtWebEngineProc                                                                                                                             
      1 root      20   0  175292  11468   8652 S   0.3   0.1   1:01.60 systemd                                                                                                                                     
    588 enrico    20   0 5147140 238336  67700 S   0.3   3.0   2:55.73 dropbox                                                                                                                                     
    592 enrico    20   0  356232  27048  15888 S   0.3   0.3  14:49.91 compton                                                                                                                                     
  31533 enrico    20   0    3984   1012    852 S   0.3   0.0   0:21.89 xautolock                                                                                                                                   
 243466 enrico    20   0   27396  16364  11836 S   0.3   0.2   0:00.08 urxvt                                                                                                                                       
      2 root      20   0       0      0      0 S   0.0   0.0   0:00.05 kthreadd                                                                                                                                    
      3 root       0 -20       0      0      0 I   0.0   0.0   0:00.00 rcu_gp                                                                                                                                      
      4 root       0 -20       0      0      0 I   0.0   0.0   0:00.00 rcu_par_gp                                                                                                                                  
      6 root       0 -20       0      0      0 I   0.0   0.0   0:00.00 kworker/0:0H-kblockd                                                                                                                        
      8 root       0 -20       0      0      0 I   0.0   0.0   0:00.00 mm_percpu_wq                                                                                                                                
      9 root      20   0       0      0      0 S   0.0   0.0   0:12.68 ksoftirqd/0                                                                                                                                 
     10 root      -2   0       0      0      0 S   0.0   0.0   0:00.00 rcuc/0                                                                                                                                      
     11 root      -2   0       0      0      0 I   0.0   0.0   0:58.57 rcu_preempt                                                                                                                                 
     12 root      -2   0       0      0      0 S   0.0   0.0   0:00.00 rcub/0                                                                                                                                      
     13 root      rt   0       0      0      0 S   0.0   0.0   0:00.23 migration/0                                                                                                                                 
     14 root     -51   0       0      0      0 S   0.0   0.0   0:00.00 idle_inject/0                                                                                                                               
     16 root      20   0       0      0      0 S   0.0   0.0   0:00.00 cpuhp/0                                                                                                                                     
     17 root      20   0       0      0      0 S   0.0   0.0   0:00.00 cpuhp/1                                                                                                                                     
     18 root     -51   0       0      0      0 S   0.0   0.0   0:00.00 idle_inject/1                                                                                                                               
     19 root      rt   0       0      0      0 S   0.0   0.0   0:00.49 migration/1                                                                                                                                 
     20 root      -2   0       0      0      0 S   0.0   0.0   0:00.00 rcuc/1          
```
